[libbeat] crashes when the eventMap contains an implementation of type *encoding.TextMarshaler and value nil

Version: libbeat any version (tested against mater 264e7b428389e73246deaf05913f274cf6fd5e2b)

**Steps to reproduce**

```
var completed *time.Time // not yet completed
event := common.MapStr{
    "date.completed": completed
}
esclient.Publish(event)
```

**Expected result**

The `date.completed` field is dropped similar to other nil pointers

**Actual result**
```
panic: value method time.Time.MarshalText called using nil *Time pointer [recovered]
        panic: value method time.Time.MarshalText called using nil *Time pointer
```

**Reason:**
`time.Time` implements `encoding.TextMarshaler`. therefore it goes to a specific case that doesn't check for nil values

--- a/libbeat/common/event.go
+++ b/libbeat/common/event.go
@@ -158,7 +158,7 @@ func normalizeValue(value interface{}, keys ...string) (interface{}, []error) {
 
 	switch value.(type) {
 	case encoding.TextMarshaler:
-		if reflect.ValueOf(value).IsNil() {
+		if reflect.ValueOf(value).Kind() == reflect.Ptr && reflect.ValueOf(value).IsNil() {
 			return nil, nil
 		}
 		text, err := value.(encoding.TextMarshaler).MarshalText()

--- a/libbeat/common/event.go
+++ b/libbeat/common/event.go
@@ -158,6 +158,9 @@ func normalizeValue(value interface{}, keys ...string) (interface{}, []error) {
 
 	switch value.(type) {
 	case encoding.TextMarshaler:
+		if reflect.ValueOf(value).IsNil() {
+			return nil, nil
+		}
 		text, err := value.(encoding.TextMarshaler).MarshalText()
 		if err != nil {
 			return nil, []error{errors.Wrapf(err, "key=%v: error converting %T to string", joinKeys(keys...), value)}

--- a/libbeat/common/event_test.go
+++ b/libbeat/common/event_test.go
@@ -198,6 +198,7 @@ func TestNormalizeValue(t *testing.T) {
 	logp.TestingSetup()
 
 	var nilStringPtr *string
+	var nilTimePtr *time.Time
 	someString := "foo"
 
 	type mybool bool
@@ -211,6 +212,7 @@ func TestNormalizeValue(t *testing.T) {
 		{nil, nil},
 		{&someString, someString},   // Pointers are dereferenced.
 		{nilStringPtr, nil},         // Nil pointers are dropped.
+		{nilTimePtr, nil},           // Nil pointers are dropped.
 		{NetString("test"), "test"}, // It honors the TextMarshaler contract.
 		{true, true},
 		{int8(8), int8(8)},

--- a/libbeat/common/event_test.go
+++ b/libbeat/common/event_test.go
@@ -22,6 +22,7 @@ import (
 	"testing"
 	"time"
 
+	uuid "github.com/satori/go.uuid"
 	"github.com/stretchr/testify/assert"
 
 	"github.com/elastic/beats/libbeat/logp"
@@ -200,6 +201,7 @@ func TestNormalizeValue(t *testing.T) {
 	var nilStringPtr *string
 	var nilTimePtr *time.Time
 	someString := "foo"
+	uuidValue := uuid.NewV1()
 
 	type mybool bool
 	type myint int32
@@ -210,10 +212,11 @@ func TestNormalizeValue(t *testing.T) {
 		out interface{}
 	}{
 		{nil, nil},
-		{&someString, someString},   // Pointers are dereferenced.
-		{nilStringPtr, nil},         // Nil pointers are dropped.
-		{nilTimePtr, nil},           // Nil pointers are dropped.
-		{NetString("test"), "test"}, // It honors the TextMarshaler contract.
+		{&someString, someString},       // Pointers are dereferenced.
+		{nilStringPtr, nil},             // Nil pointers are dropped.
+		{nilTimePtr, nil},               // Nil pointers are dropped.
+		{uuidValue, uuidValue.String()}, // Struct that honors the TextMarshaler contract.
+		{NetString("test"), "test"},     // It honors the TextMarshaler contract.
 		{true, true},
 		{int8(8), int8(8)},
 		{uint8(8), uint8(8)},

